config/core/300-resources/service.yaml

```diff
@@ -32,11 +32,30 @@ spec:
     schema:
       openAPIV3Schema:
         type: object
-        # this is a work around so we don't need to flush out the
-        # schema for each version at this time
-        #
-        # see issue: https://github.com/knative/serving/issues/912
         x-kubernetes-preserve-unknown-fields: true
+        properties:
+          spec:
+            type: object
+            x-kubernetes-preserve-unknown-fields: true
+            properties:
+              template:
+                x-kubernetes-preserve-unknown-fields: true
```

Comment:
Does this (and traffic) lack `type: object` maybe?

Reply:
According to the docs, you shouldn't need that if you have the flag I used. But I added it just to see if it would help.
For `Traffic` it is an array of objects, so set it that way in the latest commit.